pki/ssh/format.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package ssh
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/juju/errors"
+	cryptossh "golang.org/x/crypto/ssh"
+)
+
+func encodePrivateKey(pk crypto.PrivateKey) (string, error) {
+	pkcs, err := x509.MarshalPKCS8PrivateKey(pk)
+	if err != nil {
+		return "", err
+	}
+	block := &pem.Block{
+		Type:  "PRIVATE KEY",
```

Comment:
Is this correct? I thought different types of keys had different PEM blocks starts like `EC Key` etc. Does this function also make sense in a more general form somewhere else in the package where multiple things can use it. Not sure if ssh specific.

Reply:
`PRIVATE KEY` is PKCS8 encoded which should support everything. AFAIK it only depends on OpenSSH being linked/compiled with OpenSSL